includes/Handlers/Connection.php

```diff
@@ -313,6 +320,11 @@ public function handle_connect() {
 
 			facebook_for_woocommerce()->log( sprintf( 'Connection failed: %s', $exception->getMessage() ) );
 
+			set_transient( 'wc_facebook_connection_failed', time(), 30 );
+		} catch ( Connect_WC_API_Exception $exception ) {
+
+			facebook_for_woocommerce()->log( sprintf( 'Failed to connect to Facebook. No %s provided', $exception->getMessage() ), 'facebook_for_woocommerce_connect' );
```

Comment:
Is that message make sense for all possible error codes? I'd make it general, for example `Failed to connect to Facebook. Facebook API returned error code %s`

Reply:
updated in 31d7649